web-registry/src/hooks/useQueryProjects.ts

```diff
@@ -0,0 +1,34 @@
+import { useState, useEffect } from 'react';
+
+import {
+  QueryClientImpl,
+  QueryProjectsResponse,
+} from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+
+import { useLedger } from '../ledger';
+
+export default function useQueryProjects(): QueryProjectsResponse | undefined {
+  const { api } = useLedger();
+  const [queryClient, setQueryClient] = useState<QueryClientImpl>();
+  const [projects, setProjects] = useState<QueryProjectsResponse | undefined>(
+    undefined,
+  );
+
+  useEffect(() => {
+    if (!api?.queryClient) return;
+    const _queryClient: QueryClientImpl = new QueryClientImpl(api.queryClient);
+    setQueryClient(_queryClient);
+  }, [api?.queryClient]);
```

Comment:
the current eslint setup probably won't like this, even though it's fine from a react standpoint

Reply:
eslint doesn't seem to complain about this